app/lib/common/account_utils.js

```diff
@@ -103,6 +114,96 @@ export default class AccountUtils {
         return fee_asset_id;
     }
 
+    static getAccountBalances(account) {
+        account =
+            !account || account.toJS ? account : ChainStore.getAccount(account);
+        let accountBalances = account.get("balances");
+        let balances = {};
+        accountBalances.forEach((balanceID, assetID) => {
+            let balanceObject = ChainStore.getObject(balanceID);
+            let balance = balanceObject
+                ? parseInt(balanceObject.get("balance"), 10)
+                : 0;
+            balances[assetID] = balance;
+        });
+        return balances;
+    }
+
+    static getPossibleFees2(account, operation, balances) {
+        let core = ChainStore.getAsset("1.3.0");
+        account =
+            !account || account.toJS ? account : ChainStore.getAccount(account);
+        if (!account || !core) {
+            return ["1.3.0"];
+        }
+        let assets = [];
+        let globalObject = ChainStore.getObject("2.0.0");
+        let fee = estimateFee(operation, null, globalObject);
+        if (Object.keys(balances).length < 1) {
+            return ["1.3.0"];
+        }
+        Object.keys(balances).forEach(key => {
+            let assetID = key;
+            let balance = balances[assetID];
+            let hasBalance = false,
+                eqFee;
+            if (assetID === "1.3.0" && balance >= fee) {
+                hasBalance = true;
+                balances[assetID] -= fee;
+            } else if (balance && ChainStore.getAsset(assetID)) {
+                let asset = ChainStore.getAsset(assetID);
+                let price = utils.convertPrice(
+                    core,
+                    asset.getIn(["options", "core_exchange_rate"]).toJS(),
+                    null,
+                    asset.get("id")
+                );
+                eqFee = parseInt(
+                    utils.convertValue(price, fee, core, asset),
+                    10
+                );
+                if (parseInt(eqFee, 10) !== eqFee) {
+                    eqFee += 1; // Add 1 to round up;
+                }
+                if (balance >= eqFee && this.checkFeePool(asset, eqFee)) {
+                    hasBalance = true;
+                    balances[assetID] -= eqFee;
+                }
+            }
+            if (hasBalance) {
+                assets.push(assetID);
+            }
+        });
+        return assets;
+    }
+
+    static getFinalFeeAsset2(
```

Comment:
What's the difference to getFinalFeeAsset?

Reply:
Although they look similar, the parameters passed in are different, and the functions called internally are also different.

Reply:
I rebased your branch as it was not compilable before. Did you dry-code this one?

Reply:
Can't compile because of a PR merge